unsloth/kernels/utils.py

```diff
@@ -60,50 +72,78 @@ def calculate_settings(n : int) -> (int, int,):
     return BLOCK_SIZE, num_warps
 pass
 
+if DEVICE_TYPE == "cuda":
+    import bitsandbytes as bnb
+    # https://github.com/bitsandbytes-foundation/bitsandbytes/pull/1330/files
+    HAS_CUDA_STREAM = Version(bnb.__version__) > Version("0.43.3")
+    get_ptr = bnb.functional.get_ptr
 
```

Comment:
get_ptr needs to be defined. Seems like 'xpu' would fall into the else category for the quantization function below in this file. As I understand it, bitsandbytes supports intel backends now. Is there a plan to integrate it? Also might be best to have "cuda" as the default option in theses cases.